test/unit/property-accessors.html

```diff
@@ -112,6 +112,22 @@
 
     })
   });
+  test('testing for deserialization of date', function(done) {
+    var dateArr = [];
+    dateArr[0] = new Date();
+    dateArr[1] = String(dateArr[0].getTime()); //string timestamp
+    dateArr[2] = dateArr[0].getTime(); //number timestamp
+    dateArr[3] = dateArr[0].toString(); //full date
+    setTimeout(function(){
+      let ser = [];
+      ser[0] = window.XFoo.prototype._deserializeValue(dateArr[1], Date);
+      ser[1] = window.XFoo.prototype._deserializeValue(dateArr[2], Date);
+      ser[2] = window.XFoo.prototype._deserializeValue(dateArr[3], Date);
```

Comment:
It seems like there is no assertion for this line? Could you add one for it? Also the variable name `ser` seems cryptic to me. Would be better to use a longer name to improve readability

Reply:
here I am just storing the return values of the deserializeValue function. I have used them in assert statements below.
I took this approach so that assert statements becomes more readable.

Reply:
I don't disagree with the separation of the `serializedValue` array and the assertions, but I meant that I don't see any reference to `ser[2]` in the assertions on the lines below.

Reply:
Ping @ronak1009 could you rebase and update the variable names per this comment 😄